src/utils/dom/selectNodesWithEq.js

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import isNonEmptyArray from "../isNonEmptyArray";
+import isNonEmptyString from "../isNonEmptyString";
+import selectNodes from "./selectNodes";
+
+// ID and CSS classes can not start with digits
+const DIGIT_IN_SELECTOR_PATTERN = /((\.|#)(-)?\d{1})/g;
+// This is required to remove leading " > " from parsed pieces
+const SIBLING_PATTERN = /^\s*>?\s*/;
+const EQ_START = ":eq(";
+const EQ_PATTERN = /:eq\((\d+)\)/g;
+
+const cleanUp = str => str.replace(SIBLING_PATTERN, "").trim();
+const isNotEqSelector = str => str.indexOf(EQ_START) === -1;
+
+const createPair = match => {
+  const first = match.charAt(0);
+  const second = match.charAt(1);
+  const third = match.charAt(2);
+  const result = { key: match };
+
+  if (second === "-") {
+    result.val = `${first}${second}\\3${third} `;
+  } else {
+    result.val = `${first}\\3${second} `;
+  }
+
+  return result;
+};
+
+export const escapeDigitsInSelector = selector => {
```

Comment:
Why not use replace?  Something like this:
```selector.replace(/(#|\.)(-)?(\d)/g, `$1$2\\$3`);```

Reply:
@jonsnyder thanks for suggestions, I have modified the code and added replace with capture groups.